draft-dss-star.md

```diff
@@ -101,200 +121,640 @@ In this document we describe one such system, namely Distributed Secret Sharing
 The following terms are used:
 
 Aggregation Server:
-: An entity that provides some tool/software, that would like to learn aggregated data points from their user-base.
+: An entity that would like to learn aggregated data from users.
 
 Randomness Server:
-: An entity that runs an oblivious pseudorandom function ({{!OPRF=I-D.irtf-cfrg-voprf}}) service that allows clients to receive pseudorandom function evaluations on their measurement and the server OPRF key, without the Randomness Server learning anything about their measurement. The clients use the output as randomness to produce the message that is then sent to the Aggregation Server.
+: An entity that runs an oblivious pseudorandom function ({{!OPRF=I-D.irtf-cfrg-voprf}})
+  service that allows clients to receive pseudorandom function evaluations on their
+  measurement and the server OPRF key, without the Randomness Server learning anything
+  about their measurement. The clients use the output as randomness to produce the
+  report that is then sent to the Aggregation Server.
+
+Anonymizing Server:
+: An entity that clients use to decouple their identity (IP address) from their
+messages sent to the Aggregation Server.
 
 Client:
-: The entity that uses the tool.
+: The entity that provides user data to the system.
 
 Measurement:
-: The unencrypted, potentially-sensitive data point that the client is asked to report.
+: The unencrypted, potentially-sensitive data that the client is asked to report.
 
-Message:
+Report:
 : The encrypted measurement being sent by the client.
 
 Auxiliary Data:
-: Arbitrary data that clients may send as part of their message, but which is only revealed when at least K encrypted measurements of the same value are received.
+: Arbitrary data that clients may send as part of their report, but which is only
+  revealed when at least K encrypted measurements of the same value are received.
 
-# System Overview
+REPORT_THRESHOLD:
+: The minimum number of reports that an Aggregation Server needs before revealing
+  client data. This value is chosen by the application.
+
+# Cryptographic Dependencies
+
+STAR depends on the following cryptographic protocols and primitives:
+
+- Threshold secret sharing (TSS); {{deps-tss}}
+- Oblivious Pseudorandom Function (OPRF); {{deps-oprf}}
+- Key Derivation Function (KDF); {{deps-kdf}}
+- Key-Committing Authenticated Encryption with Associated Data (KCAEAD); {{deps-aead}}
+
+This section describes the syntax for these protocols and primitives in more detail.
+
+## Threshold Secret Sharing {#deps-tss}
```

Comment:
Is the idea here to specify a standard secret-sharing scheme first, and then a _robust_ scheme afterwards? I was wondering if we could just go straight for robustness, which will allow us to relax authenticity below if `k` shares are provided that are valid in a group of `k + t` for some small values of `t`.

Reply:
We should define whatever scheme we need in this section. FWIW, I'm not sure we definitely need a robust scheme for all applications, but I'll leave it to y'all to sort out.